addons/mail/models/mail_message.py

```diff
@@ -1252,3 +1252,35 @@ def _invalidate_documents(self, model=None, res_id=None):
 
     def _get_search_domain_share(self):
         return ['&', '&', ('is_internal', '=', False), ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)]
+
+    def _search_messages(self, search_term, thread_model, thread_id, before=None):
+        # we replace every space by a % to avoid hard spacing matching
+        search_term = search_term.replace(" ", "%")
```

Comment:
It's searching for every word separately, regardless of the order.
We might want to tweak it later so a "perfect match" would be prioritized.

Reply:
I don't see how it would search for every word separately. Replacing spaces by `%` still means the order of the words has to be correct, just that they can be separated by more than just spaces.

If we really need any order (and I believe it would be better) each separate term should lead to an AND query that truly not dependent on order.